tutorial/managing-software.md

```diff
@@ -0,0 +1,1394 @@
+from the autoremove list so our next autoremove will not uninstall it. Let's
+test this, just to make sure!
+
+```bash
+sudo apt autoremove
+```
+
+Select <kbd>Y</kbd> when prompted this time, and then we can run
+`apt list ssl-cert` to quickly see if our `ssl-cert` package is still on the
+system:
+
+```bash
+ubuntu@tutorial:~$ apt list ssl-cert
+Listing... Done
+ssl-cert/noble,now 1.1.2ubuntu1 all [installed]
+```
+
+If you're curious, you can also run `apt list apache2` to see how the output
+differs for a package that was once installed and then removed!
+
+## Customise configuration
+
+In general, the default package configuration should just work well, and work
+"out of the box" when it's installed. But it's almost inevitable that, sooner
+or later, we'll want to customise the package so that it better fits our own
+purposes.
+
+Before we try to customise the package, we should probably look at what files
+are included in it. We can check this using `dpkg`, which is the
+[Debian package manager](https://manpages.ubuntu.com/manpages/en/man1/dpkg.1.html).
+Although APT is now more commonly used for basic package handling, `dpkg`
+retains some really helpful commands for examining files and finding out
+package information. It's installed by default on Ubuntu systems so we can use
+it directly:
+
+```bash
+dpkg --listfiles ssl-cert
+```
+
+This gives us the following list of files and their directory structure:
+
+```text
+/.
+/etc
+/etc/ssl
+/etc/ssl/certs
+/etc/ssl/private
+/lib
+diverted by base-files to: /lib.usr-is-merged
+/lib/systemd
+/lib/systemd/system
+/lib/systemd/system/ssl-cert.service
+/usr
+/usr/sbin
+/usr/sbin/make-ssl-cert
+/usr/share
+/usr/share/doc
+/usr/share/doc/ssl-cert
+/usr/share/doc/ssl-cert/README
+/usr/share/doc/ssl-cert/changelog.gz
+/usr/share/doc/ssl-cert/copyright
+/usr/share/lintian
+/usr/share/lintian/overrides
+/usr/share/lintian/overrides/ssl-cert
+[...]
+```
+
+If we find a file but we're not sure what package it comes from, `dpkg` can
+help us there too! Let's use the example of one of the files from the previous
+output: `/usr/share/ssl-cert/ssleay.cnf` and do a search for it using `dpkg`:
+
+```bash
+dpkg --search /usr/share/ssl-cert/ssleay.cnf
+```
+
+This will provide us with the package name for the given file:
+
+```bash
+ssl-cert: /usr/share/ssl-cert/ssleay.cnf
+```
+
+### Conffiles
+
+Most of a package's configuration is handled through
+[configuration files](https://www.debian.org/doc/debian-policy/ap-pkg-conffiles.html#automatic-handling-of-configuration-files-by-dpkg)
+(often known as **conffiles**).
+
+Package conffiles are different from all other files delivered in a package.
+A package may have any number of conffiles (including none!). Conffiles are
+explicitly marked by the package maintainer during development to protect them
+from being automatically upgraded, whilst all other types of files are not --
+any changes you make to regular files *will be overwritten* during an upgrade. 
```

Comment:
Doesn’t it prompt them if they want to override their existing conf files though? Do you want to note that? (I’m not actually sure if they’re always prompted–Landscape Server prompts if they want to overwrite configuration files)